src/main/webapp/placeGuides/MapPlaceGuideDisplayer.js

```diff
@@ -13,6 +13,42 @@ class MapPlaceGuideDisplayer {
     this.addNewPlaceGuidesToMap(placeGuides);
   }
 
+  remove(placeGuideId) {
+    this._placeGuidesOnMap[placeGuideId].remove();
+    delete this._placeGuidesOnMap[placeGuideId];
+  }
+
+  highlight(placeGuideId) {
+    this._placeGuidesOnMap[placeGuideId].highlight();
+  }
+
+  unhighlight(placeGuideId) {
+    this._placeGuidesOnMap[placeGuideId].unhighlight();
+  }
+
+  // This function sets the map bound to the minimum one
+  // which contains all the PlaceGuides.
+  adjustMapToShowAll() {
+    let mapBounds = new google.maps.LatLngBounds();
+    let guidesExist = false;
+    const placeGuideIds = Object.getOwnPropertyNames(this._placeGuidesOnMap);
+    placeGuideIds.forEach((placeGuideId) => {
+        const position =
+          this._placeGuidesOnMap[placeGuideId].marker.getPosition();
+        mapBounds.extend(position);
+        guidesExist = true;
+      }
+    );
+    if (guidesExist) {
```

Comment:
I think you can use "placeGuideIds.length > 0" here instead of managing the variable.